src/Nethermind/Nethermind.Blockchain/TransactionPools/ITransactionPool.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Model;
+using Nethermind.Dirichlet.Numerics;
+
+namespace Nethermind.Blockchain.TransactionPools
+{
+    public interface ITransactionPool
+    {
+        Transaction[] PendingTransactions { get; }
+        TransactionReceipt GetReceipt(Keccak hash);
+        void AddFilter<T>(T filter) where T : ITransactionFilter;
```

Comment:
it might be a functionality that is added to early - for now I would prefer to support only adding filters - see no reason for removing them dynamically yet - we will add when confirmed that is needed